### ISSUE TITLE
Refactoring hedera services CircleCi build scripts 

### ISSUE THREAD

**Issue by ljianghedera** [hashgraph/hedera-services#241]
Since it's difficulty to achieve 0 failure regression workflow given the current situation, we'd like to enhance the hedera services CircleCi nightly regression to achieve the following: 

- [ ] Only fail the nightly regression workflow if the process can't move forward, for example, if the AWS test network is not available. Otherwise, we shall keep the workflow till the end;

- [ ] If any of the worker test-client fails, don't fail the overall workflow;

- [ ] Further, for any worker test-client failure, don't report to slack;

- [ ] For the failed worker test-client, record what type of failure and if possible, how many of the failures, etc. to provide the stats for the final slack report;

- [ ] Only at the end of the workflow, final report with summary and stats will be created and sent to hedera-regression slack channel just once for each regression test run;

In summary, hedera services nightly regression shall follow platform regression's example.
